### PR TITLE
Implemented Elvis ValueExpression (#34)

### PR DESCRIPTION
**Pull request by rdvdijk**

### CODE REVIEW

File: src/main/java/io/parsingdata/metal/expression/value/Elvis.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public OptionalValue eval(final Environment env, final Encoding enc) {
+        final OptionalValue eval = _l.eval(env, enc);
+        if (eval.isPresent()) {
+            return eval;
+        }
+        return _r.eval(env, enc);
+    }
+}
```
Comment (jvdb) [inline src/main/java/io/parsingdata/metal/expression/value/Elvis.java]: toString() is missing!

Comment (rdvdijk) [inline src/main/java/io/parsingdata/metal/expression/value/Elvis.java]: Indeed! I've added `toString()`, a unit test for it, and also added it to the `ToStringTest`.

File: src/main/java/io/parsingdata/metal/expression/value/Elvis.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private final ValueExpression _lop;
+    private final ValueExpression _rop;
+
+    public Elvis(final ValueExpression lop, final ValueExpression rop) {
+        _lop = lop;
+        _rop = rop;
+    }
+
+    @Override
+    public OptionalValue eval(final Environment env, final Encoding enc) {
```
Comment (gertjanal) [inline src/main/java/io/parsingdata/metal/expression/value/Elvis.java]: Evil Elvis? ;)

File: src/main/java/io/parsingdata/metal/Shorthand.java [hunk 3]
```diff
@@ -125,6 +126,7 @@
     public static ValueExpression offset(final String s) { return new Offset(s); }
     public static final ValueExpression currentOffset = new CurrentOffset();
     public static ValueExpression cat(final ValueExpression l, final ValueExpression r) { return new Cat(l, r); }
+    public static ValueExpression elvis(final ValueExpression l, final ValueExpression r) { return new Elvis(l, r); }
```
Comment (gertjanal) [inline src/main/java/io/parsingdata/metal/Shorthand.java]: `elv()`? :)

Comment (rdvdijk) [inline src/main/java/io/parsingdata/metal/Shorthand.java]: I expected that comment from @jvdb ;-)

**Pull request merged**